tests/FunctionalJavascript/SettingsFormTest.php

```diff
@@ -0,0 +1,273 @@
+    $assert_session->statusMessageContains('The configuration options have been saved.', 'status');
+
+    // The "never" option sets the "force_2fa" to FALSE.
+    $config = $this->loadConfig();
+    $this->assertEquals([], $config->get('2fa_conditions'));
+    $this->assertFalse($config->get('force_2fa'));
+
+    // Re-assert that the conditions are still hidden.
+    $this->drupalGet('/admin/config/system/oe_authentication');
+    $this->assertFalse($conditions_wrapper->isVisible());
+    // Set the 2FA to "always".
+    $always_radio->click();
+    $this->assertFalse($conditions_wrapper->isVisible());
+    $assert_session->buttonExists('Save configuration')->press();
+    $assert_session->statusMessageContains('The configuration options have been saved.', 'status');
+
+    // The "always" option sets the "force_2fa" to TRUE.
+    $config = $this->loadConfig();
+    $this->assertEquals([], $config->get('2fa_conditions'));
+    $this->assertTrue($config->get('force_2fa'));
+
+    $this->drupalGet('/admin/config/system/oe_authentication');
+    $this->assertFalse($conditions_wrapper->isVisible());
+    $conditional_radio->click();
+    $conditions_wrapper_selector = 'details[data-drupal-selector="edit-2fa-conditions"]';
+    $conditions_wrapper = $assert_session->waitForElementVisible('css', $conditions_wrapper_selector);
+    $this->assertNotNull($conditions_wrapper);
+
+    // Test that only condition plugins that require a user context are shown.
+    // The conditions are disabled by default.
+    $enabled_conditions_fieldset = $assert_session->elementExists('named', ['fieldset', 'Enabled conditions'], $conditions_wrapper);
+    $this->assertCount(2, $enabled_conditions_fieldset->findAll('css', 'input'));
+    $assert_session->checkboxNotChecked('User test condition', $enabled_conditions_fieldset);
+    $assert_session->checkboxNotChecked('User Role', $enabled_conditions_fieldset);
+
+    // Each plugin comes with its own configuration form.
+    $condition_settings_tabs_wrapper = $assert_session->elementExists(
+      'xpath',
+      $this->cssSelectToXpath('div.js-form-type-vertical-tabs') . '[./label[string(.)="Condition configuration"]]',
+    );
+    $this->assertCount(2, $condition_settings_tabs_wrapper->findAll('css', 'details'));
+
+    // The first condition plugin is the user test one.
+    $test_condition_wrapper = $this->assertDetailsElementExists('User test condition', $condition_settings_tabs_wrapper);
+    $this->assertFalse($assert_session->fieldExists('Example configuration option', $test_condition_wrapper)->isChecked());
+    // The second one is the user role core plugin.
+    $user_role_condition_wrapper = $this->assertDetailsElementExists('User Role', $condition_settings_tabs_wrapper);
+    $this->assertEquals([
+      'anonymous',
+      'authenticated',
+      $role_id,
+    ], array_map(
+      function (NodeElement $element): string {
+        $this->assertFalse($element->isChecked());
```

Comment:
I don't like to see this assert in the closure for array_map(). But it might be the most efficient way to do it. Ok.